src/webhooks/generic-notifier/generic-notifier.ts

```diff
@@ -0,0 +1,84 @@
+import { v1 } from '@datadog/datadog-api-client';
+import * as Sentry from '@sentry/node';
+import { FastifyReply, FastifyRequest } from 'fastify';
+import moment from 'moment-timezone';
+
+import { GenericEvent } from '@types';
+
+import { bolt } from '@/api/slack';
+import { DATADOG_API_INSTANCE } from '@/config';
+import { EVENT_NOTIFIER_SECRETS } from '@/config/secrets';
+import { extractAndVerifySignature } from '@/utils/auth/extractAndVerifySignature';
+
+export async function genericEventNotifier(
+  request: FastifyRequest<{ Body: GenericEvent }>,
+  reply: FastifyReply
+): Promise<void> {
+  try {
+    // If the webhook secret is not defined, throw an error
+    const { body }: { body: GenericEvent } = request;
+    if (
+      body.source === undefined ||
+      EVENT_NOTIFIER_SECRETS[body.source] === undefined
+    ) {
+      reply.code(400).send('Invalid source or missing secret');
+      throw new Error('Invalid source or missing secret');
+    }
+
+    const isVerified = await extractAndVerifySignature(
+      request,
+      reply,
+      'x-infra-hub-signature',
+      EVENT_NOTIFIER_SECRETS[body.source]
+    );
+    if (!isVerified) {
+      // If the signature is not verified, return (since extractAndVerifySignature sends the response)
+      return;
+    }
+
+    await messageSlack(body);
+    await sendEventToDatadog(body, moment().unix());
+    reply.code(200).send('OK');
+    return;
+  } catch (err) {
+    console.error(err);
+    Sentry.captureException(err);
+    reply.code(500).send();
+    return;
+  }
+}
+
+export async function sendEventToDatadog(
+  message: GenericEvent,
+  timestamp: number
+) {
+  if (message.data.channels.datadog) {
```

Comment:
I'd like a pattern where you decide which methods to call in the `genericEventNotifier` function. That is the orchestrating function, which parses the message and trigger the relevant channels. From their name `sendEventToDatadog` and `messageSlack` should always send the message, not decide when not to send.